src/main/java/pl/coderstrust/accounting/logic/pdf/PdfService.java

```diff
@@ -0,0 +1,179 @@
+    nextTextY -= table.getRowHeight();
+    nextTextX = table.getMargin() + table.getCellMargin();
+
+    for (int i = 0; i < currentPageContent.length; i++) {
+      writeContentLine(currentPageContent[i], contentStream, nextTextX, nextTextY, table);
+      nextTextY -= table.getRowHeight();
+      nextTextX = table.getMargin() + table.getCellMargin();
+    }
+    contentStream.close();
+  }
+
+  private static void writeContentLine(String[] lineContent, PDPageContentStream contentStream,
+      float nextTextX, float nextTextY,
+      Table table) throws IOException {
+    for (int i = 0; i < table.getNumberOfColumns(); i++) {
+      String text = lineContent[i];
+      contentStream.beginText();
+      contentStream.newLineAtOffset(nextTextX, nextTextY);
+      contentStream.showText(text != null ? text : "");
+      contentStream.endText();
+      nextTextX += table.getColumns().get(i).getWidth();
+    }
+  }
+
+  private static void drawTableGrid(Table table, String[][] currentPageContent,
+      PDPageContentStream contentStream, float tableTopY)
+      throws IOException {
+    float nextY = tableTopY;
+    for (int i = 0; i <= currentPageContent.length + 1; i++) {
+      contentStream.moveTo(table.getMargin(), nextY);
+      contentStream.lineTo(table.getMargin() + table.getWidth(), nextY);
+      contentStream.stroke();
+      nextY -= table.getRowHeight();
+    }
+
+    final float tableYLength = table.getRowHeight() + (table.getRowHeight() * currentPageContent.length);
+    final float tableBottomY = tableTopY - tableYLength;
+    float nextX = table.getMargin();
+    for (int i = 0; i < table.getNumberOfColumns(); i++) {
+      contentStream.moveTo(nextX, tableTopY);
+      contentStream.lineTo(nextX, tableBottomY);
+      contentStream.stroke();
+      nextX += table.getColumns().get(i).getWidth();
+    }
+    contentStream.moveTo(nextX, tableTopY);
+    contentStream.lineTo(nextX, tableBottomY);
+    contentStream.stroke();
+  }
+
+  private static String[][] getContentForCurrentPage(Table table, Integer rowsPerPage,
+      int pageCount) {
+    int startRange = pageCount * rowsPerPage;
+    int endRange = (pageCount * rowsPerPage) + rowsPerPage;
+    if (endRange > table.getNumberOfRows()) {
+      endRange = table.getNumberOfRows();
+    }
+    return Arrays.copyOfRange(table.getContent(), startRange, endRange);
+  }
+
+  private static PDPage generatePage(PDDocument doc, Table table) {
+    PDPage page = new PDPage();
+    page.setMediaBox(table.getPageSize());
+    page.setRotation(table.isLandScape() ? 90 : 0);
+    doc.addPage(page);
+    return page;
+  }
+
+  private static PDPageContentStream generateContentStream(PDDocument doc, PDPage page, Table table, Invoice invoice)
+      throws IOException {
+
+    PDPageContentStream contentStream = new PDPageContentStream(doc, page, AppendMode.APPEND, false);
+    contentStream.beginText();
```

Comment:
this method is too long and not readable - please split it into few methods and just call those methods here - e.g. addSeller(), addBuyer() etc.

Reply:
done